Home: Mobile setup tasks reference the Jetpack app

dcalhoun opened the pull request:
#### Proposed Changes

Following the plans to [refocus the WordPress apps on core features](https://make.wordpress.org/mobile/2022/07/27/refocusing-the-wordpress-app-on-core-features/), we plan to move features that are specific to WordPress.com to the Jetpack app. As such, we will eventually replace all mentions of the WordPress app in Calypso with Jetpack.

The following changes only apply to mobile device; larger device continue to see the current experience. A PR for larger devices will arrive later. 

* Add a Jetpack app overlay to _Home_. 
* Remove one of two app CTAs found on the home page for mobile web browsers. 
* Update copy/links for the retained app CTA to focus on the Jetpack app. 


<details><summary>UX Video</summary>

https://user-images.githubusercontent.com/438664/191395653-850b179d-cb92-4594-9331-ea218d5f8f5f.mov

</details>

<details><summary>Screenshots</summary>

| Before (Mobile) | After (Mobile) |
| - | - |
| ![home-mobile-before](https://user-images.githubusercontent.com/438664/191369006-258fc510-f04f-4983-bde5-c3643a7fc975.png) | ![home-mobile-after](https://user-images.githubusercontent.com/438664/191395330-f692ee48-e71f-42d8-a4dc-845d8e9016b1.png) |
| ![home-mobile-before-2](https://user-images.githubusercontent.com/438664/191369005-fa77d4aa-82c0-4876-86f7-ec676450b82e.png) |  <img width="376" alt="home-mobile-after-2" src="https://user-images.githubusercontent.com/438664/192872336-74a79556-2087-40d7-a0cf-5a2134ea8e8c.png"> |
| | ![home-mobile-overlay](https://user-images.githubusercontent.com/438664/191369007-68773a0c-5c1f-498a-adfc-93b1907220c8.png) |

| Before (Desktop) | After (Desktop) |
| - | - |
| ![home-desktop-before](https://user-images.githubusercontent.com/438664/191369001-c589c9b2-adac-4e80-b544-38ff729dcf29.png) | ![home-desktop-after](https://user-images.githubusercontent.com/438664/191392235-3e22d8f7-f51a-4865-8ed8-7f25dde3ba7b.png) |

</details>

#### Testing Instructions

* Verify the _Home_ overlay displays for mobile web browsers only and links to the Jetpack app.<sup>1</sup>
* Verify the only one app CTA is displayed in mobile web browsers and links to the Jetpack app. 
* Verify the app CTA on non-mobile web browsers remains the same as production and links to the WordPress mobile app. 

<sup>1. If the banner has already been dismissed, you can run `dispatch( { type: 'PREFERENCES_SET', key: 'appBannerDismissTimes', value: null } );` in the browser console. In order to run this in the console, you must use a local development server or append `?debug` to the URL.</sup>

#### Pre-merge Checklist

- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [x] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?

dcalhoun committed:
Revert "Add: Mobile app banner to site task list (#58612)"

This reverts commit 19e134305ffb1182d511943870ee216a59379cdd.

dcalhoun committed:
feat: My Sites mobile CTA points to Jetpack app

We want to direct users to install the Jetpack app instead of the
WordPress app, as we focus on improving the former. This change is
currently scoped to mobile web browsers, as we have not updated all the
CTAs displayed for non-mobile web browsers; we do not want to display
conflicting CTAs.

The button to email oneself an app install link is replaced with a link
to the Jetpack marketing site as it is not obvious the link would send
an email, it would require additional copy to explain. Instead, we
direct to the Jetpack marketing site containing additional details on
the mobile app.

dcalhoun committed:
feat: Add Calypso app banner to Home

Display a banner now that the permanent call out towards the top of the
page is removed.

dcalhoun committed:
feat: Hide app setup task in mobile web browsers

This CTA still points to the WordPress mobile app. Displaying it would
conflict with Jetpack app CTAs. It will be updated when we focus upon
non-mobile web browsers.

dcalhoun commented:
Hey @shaunandrews. 👋  I am interested in your review from a design perspective. There is a UX recording and screenshots included in the PR description. Please let me know if anything extra is needed.

SiobhyB commented on client/blocks/app-banner/utils.js:
```diff
@@ -43,12 +53,20 @@ export function getAppBannerData( translate, sectionName, isRTL ) {
+				};
+			}
+
+			return emptyBanner;
```
When I set `displayJetpackAppBranding` to false, I see that the banner is now showing on the home page, but with no content:

<img width="320" alt="Screenshot 2022-09-21 at 16 36 36" src="https://user-images.githubusercontent.com/2998162/191548467-e19f0ff4-d522-429e-992f-fe6f2088331f.png">

If there isn't a simple way to hide this in production for now, maybe we could update the default text to something more generic?

dcalhoun replied:
Thank you for catching this. Addressed in 30b19ab.

SiobhyB commented on client/blocks/app-banner/index.jsx:
```diff
@@ -134,6 +135,8 @@ export class AppBanner extends Component {
 			switch ( currentSection ) {
 				case GUTENBERG:
 					return `intent://post/#Intent;scheme=${ scheme };package=${ packageName };end`;
+				case HOME:
```
For this to work, I believe we'll need to add `home` [here in the Android app's manifest](https://github.com/wordpress-mobile/WordPress-Android/blob/55bb3298d72aadc5850bc0ee2e5ce03a2f4136fe/WordPress/src/jetpack/AndroidManifest.xml#L24-L38). A deep link to the home section isn't currently in the WordPress app either (judging from [this list](https://github.com/wordpress-mobile/WordPress-Android/blob/a96b4ac411f55e287665cc643bcbb31f970ed4eb/WordPress/src/main/AndroidManifest.xml#L80-L91)), so we'll likely also need to figure out a way to link the intent to the app's home page. This isn't a blocker to this PR, but I wanted to note it as something we'll need to work on down the road.

dcalhoun replied:
Aha. Thank you for sharing this. This makes sense. I will add this to my todo list to manage. 🙇🏻 

In theory, I suppose only _need_ to add this for the Jetpack app, unless we just want the ability for WordPress app. There is not a banner linking to WordPress app's home to my knowledge.

dcalhoun replied:
For posterity, https://github.com/wordpress-mobile/WordPress-Android/pull/17239 adds this `home` intent.

SiobhyB reviewed (changes requested):
Thanks @dcalhoun! I tested and the changes look great overall to me. 

I've added a couple of small comments about a future task we'll need to work on (not a blocker for this PR) and the appearance of the banner when `displayJetpackAppBranding` isn't enabled. 

For the latter point, if there isn't a straightforward way to remove the banner on the home page in production, I think it'd be okay to keep it and add some generic text about the app.

dcalhoun committed:
fix: Prevent display of an empty banner

Because the `HOME` section is only used when the relevant feature flag
is enabled, `getAppBannerData` is capable of returning empty title and
copy values. However, the `ALLOWED_SECTIONS` is a static assignment, so
this resulted in the app banner erroneously displaying with no title or
copy for the `HOME` section when the feature flag was disabled.

SiobhyB reviewed (approved):
Thanks for the updates, re-tested and looks good from my perspective :) 🙌

dcalhoun commented on client/blocks/app-banner/index.jsx:
```diff
@@ -197,6 +200,13 @@ export class AppBanner extends Component {
 		const { title, copy } = getAppBannerData( translate, currentSection );
 
+		// This conditional will be unnecessary once the 'jetpack/app-branding'
+		// feature flag is removed and its features re made the default
```
```suggestion
		// feature flag is removed and its features are made the default
```

dcalhoun committed:
fix: Typo in comment

shaunandrews commented:
Can you confirm this will receive the updates outlined over in #67607, specifically the font-weight and spacing changes?

--

Can we add a gray dividing line above the app callout? Right now it feels too visually connected to the Quick Links section.

dcalhoun committed:
feat: Add separator between Quick Links and Mobile CTA

Increase visual separation between the two segments. Leverage existing
CSS selector to apply the styles.

dcalhoun committed:
Merge branch 'trunk' of github.com:Automattic/wp-calypso into update/mobile-setup-tasks-reference-jetpack-app

dcalhoun commented:
> Can you confirm this will receive the updates outlined over in #67607, specifically the font-weight and spacing changes?

@shaunandrews yes, it has received the updates from #67607 now that I have pulled the latest `trunk` branch into this branch. 

> Can we add a gray dividing line above the app callout? Right now it feels too visually connected to the Quick Links section.

I added a dividing line leveraging an existing CSS selector found in the code base. I updated the mobile screenshot, the non-mobile UI was not impacted. 

<details><summary>UI w/ divider</summary>

| Before (Mobile) | After (Mobile) |
| - | - |
| <img width="376" alt="home-mobile-before-2" src="https://user-images.githubusercontent.com/438664/191369005-fa77d4aa-82c0-4876-86f7-ec676450b82e.png"> |  <img width="376" alt="home-mobile-after-2" src="https://user-images.githubusercontent.com/438664/192872336-74a79556-2087-40d7-a0cf-5a2134ea8e8c.png"> |

</details>

a8ci18n commented:
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/7556591

Thank you @dcalhoun for including a screenshot in the description! This is really helpful for our translators.

shaunandrews commented:
Can we adjust some of the spacing [to match the Figma design](https://www.figma.com/file/qk5s4q4mZsXknwhMWo2vIY/WordPress--%3E-Jetpack-Migration?node-id=1428%3A6930) closer. Here's a comparison with the design on the left, and your screenshot on the right:

<img width="1318" alt="CleanShot 2022-09-30 at 14 30 17@2x" src="https://user-images.githubusercontent.com/191598/193334275-e8ad7c16-9e34-445c-ad7a-875a97a0f961.png">

dcalhoun commented:
> Can we adjust some of the spacing [to match the Figma design](qk5s4q4mZsXknwhMWo2vIY-fi-1428%3A6930) closer.

Sure thing. I will open a new PR improving the spacing.

a8ci18n commented:
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/7556591).
